### DISCUSSION TITLE
Calculating reward from state representation

### DISCUSSION THREAD

**Discussion by kelemenr** [eleurent/highway-env#297] [General]
Dear Edouard,
I would like to ask if you could help me find a way of how I could calculate the reward (not in parking, but in the other environments, mostly highway-v0) from a predicted next state representation?
Best Regards,
R.

**Comment by eleurent** [comment]
The reward is computed from the true environment state (position, speed, internal state etc. of every vehicle in the scene + road network), and this computation cannot be performed easily from a state representation (it would require to reconstruct the environment state from this representation first).
So the easiest solution is probably to train a model to predict the reward r(s,a,s')=r(s') from a state representation h(s'), given a dataset of transitions (s, a, r, s').
Then, this model can be used in addition to a dynamics model: from current state s and action a, predict the next state representation h(s') and associated reward R(h(s')).
